include/gatekeeper_cps.h

```diff
@@ -46,10 +50,67 @@ struct cps_config {
 	uint16_t          tx_queue_front;
 	uint16_t          rx_queue_back;
 	uint16_t          tx_queue_back;
+
+	/* ACL used for classifying packets coming from a KNI. */
+	struct rte_acl_ctx *kni_acl;
+
+	/* Unanswered resolution requests from the KNIs. */
+	struct list_head  arp_requests;
+	struct list_head  nd_requests;
+
+	/* Timer to scan over outstanding resolution requests. */
+	struct rte_timer  scan_timer;
+};
+
+/* Information needed to submit IPv6 BGP packets to the CPS block. */
+struct cps_bgp_req {
+	/* IPv6 BGP packets. */
+	struct rte_mbuf      *pkts[GATEKEEPER_MAX_PKT_BURST];
+
+	/* Number of packets stored in @pkts. */
+	unsigned int         num_pkts;
+
+	/* KNI that should receive @pkts. */
+	struct rte_kni       *kni;
+};
+
+/* Information needed to submit ARP responses to the CPS block. */
+struct cps_arp_req {
```

Comment:
The comment should be more informative since this request is only informing the CPS block to remove an entry from the watch list, that is, it not sending an ARP reply. The same goes for the IPv6 case.